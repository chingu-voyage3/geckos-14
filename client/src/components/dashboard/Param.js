/** Param Component:
Param component is the smallest component of the controlPanel.
TODO: Allow the composistion of different type of input (select, checkbox, radio)
*/
import React from 'react';

const Param = props => {
  return (
    <label className="param-label" htmlFor={'param-id-' + props.name}>
      {props.label}
      <input
        className={'param-' + props.name}
        name={props.name}
        value={props.value}
        onChange={props.onChange}
        id={'param-id-' + props.name}
      />
    </label>
  );
};

export default Param;
