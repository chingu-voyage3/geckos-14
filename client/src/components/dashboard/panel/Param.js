/** Param Component:
Param component is the smallest component of the controlPanel.
TODO: Allow the composistion of different type of input (select, checkbox, radio)
*/
import React from 'react';

// render Method for Select input
const renderSelect = options => {
  return options.map(option => {
    // TODO: Add in option objects value and display text
    return (
      <option key={'SD-' + option.id} value={option.value}>
        {option.display}
      </option>
    );
  });
};
// redering specific type for each Param.
// TODO: Transform to switch with different type if needed

const Param = props => {
  return (
    <div className="control-panel-param">
      <label htmlFor={'param-id-' + props.name}>{props.label}</label>
      {props.type === 'select' ? (
        <select className="" {...props}>
          {renderSelect(props.options)}
        </select>
      ) : (
        <input className="" {...props} />
      )}
    </div>
  );
};

export default Param;
