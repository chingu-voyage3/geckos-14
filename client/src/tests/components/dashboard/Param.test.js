import React from 'react';
import Param from '../../../components/dashboard/Param';

import { shallow } from 'enzyme';
const func = e => {
  console.log(e.target.value);
};
const param = {
  label: 'label',
  value: 'value',
  name: 'name',
  onChange: func
};
it('renders without crashing', () => {
  const div = document.createElement('div');
  const wrParamer = shallow(<Param {...param} />);
  expect(toJson(wrParamer)).toMatchSnapshot();
});
