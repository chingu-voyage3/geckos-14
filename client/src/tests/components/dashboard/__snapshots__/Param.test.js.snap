// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<label
  className="control-panel-param"
  htmlFor="param-id-name"
>
  label
  <input
    label="label"
    name="name"
    onChange={[Function]}
    value="value"
  />
</label>
`;
